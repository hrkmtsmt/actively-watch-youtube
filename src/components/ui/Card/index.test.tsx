import React from 'react';
import { describe, test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Card } from '.';
import '@testing-library/jest-dom/vitest';

describe('Button component', () => {
  test('snapshot', () => {
    render(
      <Card
        title="The Pragmatic Programmer"
        url="https://example.com"
        image={{ url: 'https://example.com/image', width: 256, height: 256 }}
      />
    );
    expect(document.body.getElementsByTagName('a')[0]).toMatchSnapshot();
  });

  test('render dom with children.', () => {
    render(
      <Card
        title="Test Driven Development"
        url="https://example.com"
        image={{ url: 'https://example.com/image', width: 256, height: 256 }}
      />
    );
    expect(screen.getByRole('link')).toHaveTextContent('Test Driven Development');
    expect(screen.getByRole('link')).toHaveAttribute('href', 'https://example.com');
    expect(screen.getByRole('img')).toHaveAttribute('alt', 'Test Driven Development');
    expect(screen.getByRole('img')).toHaveAttribute('src', 'https://example.com/image');
  });
});
