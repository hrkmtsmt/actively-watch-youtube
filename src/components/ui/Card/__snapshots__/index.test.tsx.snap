// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > snapshot 1`] = `
<a
  class="style_wrapper__seat0h0"
  href="https://example.com"
  title="The Pragmatic Programmer"
>
  <img
    alt="The Pragmatic Programmer"
    class="style_image__seat0h1"
    height="256"
    src="https://example.com/image"
    width="256"
  />
  <div
    class="style_body__seat0h2"
  >
    <h3
      class="style_title__seat0h3"
    >
      The Pragmatic Programmer
    </h3>
  </div>
</a>
`;
